### PR TITLE
docs/how-to: update doc for titan

### PR DESCRIPTION
**Pull request by dcalvin**
## What have you changed? (mandatory)

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:
-  add document for Titan. Original Jira ticket: 
https://internal.pingcap.net/jira/browse/TIKV-3577

## What are the type of the changes? (mandatory)

The currently defined types are listed below, please pick one of the types for this PR by removing the others:
- Doc update

## Does this PR affect documentation (docs) or release note? (mandatory)

- yes

### CODE REVIEW

File: docs/how-to/configure/titan.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+To enable Titan in TiKV:
+
+In the TiKV configuration file, specify `rocksdb.titan.enabled=true`.
+
+## How to fall back to RocksDB
+
+If you find Titan does not help or is causing read or other performance issues, you can take the following steps to fall back to RocksDB:
+
+1. Enter the `tikv-ctl` interface and specify `blob-run-mode = “fallback” / “read-only”`.
```
Comment (Hoverbear) [inline docs/how-to/configure/titan.md]: Let's make this a command we can run! It's not discoverable or usable like this.

I spoke with @yiwu-arbug 

```shell
tikv-ctl --host 127.0.0.1:20160 modify-tikv-config -m kvdb -n default.blob_run_mode -v "kFallback"
```

We should also note if titan is not enabled this command is not available if `rocksdb.titan.enabled=false`.

File: docs/how-to/configure/titan.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+## How Titan works
+
+![Titan Architecture](../../images/titan-architeture)
+
+Titan seperates values from the LSM-tree during flush and compaction. While the actual value is stored in a blob file, the value in the LSM tree functions as the position index of the actual value. When a GET operation is performed, Titan obtains the blob index for the corresponding key from the LSM tree. Using the index, Titan identifies the actual value from the blob file and returns it. For more details on design and implementation of Titan, please refer to [Titan: A RocksDB Plugin to Reduce Write Amplification](https://pingcap.com/blog/titan-storage-engine-design-and-implementation/).
+
+> Caveat:
+>
+> Titan's improved write performance is at the cost of sacrificing storage space and range query performance. It's only recommended for scenarios of large values (>= 1K).
```
[the remark this answers is not in the document]
Comment (Connor1996) [reply docs/how-to/configure/titan.md]: Maybe `It's only recommended for ...` is too absolute, how about `Mostly we recommend it for ...`?

File: docs/how-to/configure/titan.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+---
+title: Titan
+summary: Learn how to enable Titan in TiKV.
+category: how-to
+---
+
+# Titan
+
+Titan is an alternative key-value storage engine developed by PingCAP. It is based on RocksDB, but can reduce write amplification in RocksDB when using large values.
```
Comment (yiwu-arbug) [inline docs/how-to/configure/titan.md]:
```suggestion
Titan is an plugin of RocksDB developed by PingCAP to provide key-value separation, inspired by WiscKey. The goal of Titan is to reduce write amplification of RocksDB when using large values.
```

Comment (yiwu-arbug) [reply docs/how-to/configure/titan.md]: A "storage engine" is a strong claim. A plugin of RocksDB is a more proper description of Titan.

File: docs/how-to/configure/titan.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (dcalvin) [reply docs/how-to/configure/titan.md]: > Maybe `It's only recommended for ...` is too absolute, how about `Mostly we recommend it for ...`?

OK. Also please confirm the value size limit ">=1KB".

File: docs/how-to/configure/titan.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+>
+> Titan's improved write performance is at the cost of sacrificing storage space and range query performance. It's only recommended for scenarios of large values (>= 1K).
+
+## How to enable Titan
+
+As Titan has not reached an appropriate maturity to be applied in production, it is disabled in TiKV by default. Before enabling it, make sure you understand the caveat as mentioned above and that you have evaluated your scenarios and needs.
+
+To enable Titan in TiKV:
+
+In the TiKV configuration file, specify `rocksdb.titan.enabled=true`.
```
[the remark this answers is not in the document]
Comment (dcalvin) [reply docs/how-to/configure/titan.md]: The note info (cannot fallback) in the code seems to be in conflict with our next topic **How to fall back to RocksDB**. Please advise. 
@yiwu-arbug  @Connor1996

File: docs/how-to/configure/titan.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Connor1996) [reply docs/how-to/configure/titan.md]: LGTM, PTAL @yiwu-arbug

File: docs/how-to/configure/titan.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Connor1996) [reply docs/how-to/configure/titan.md]: Oh, the config description is not updated. /cc @tabokie

**Comment by dcalvin** [comment]
@Hoverbear @Connor1996 @yiwu-arbug PTAL again and approve if there is no further issue.

**Pull request merged**